client/js/heat.js

```diff
@@ -0,0 +1,91 @@
+class HeatMapRow {
+  /**
+   * format for a row for heat map values obj
+   * @param {number[]} data The array of 0s and 1s for a row in a heat map
+   */
+  constructor(data) {
+    this.data = data;
+  }
+}
+
+// data will be from backend next time
+const TOTAL_LIKED = 13;
+// element value of x means xth latest video is music
+const LIKED_MUSIC_HIST = [1, 4, 7];
+
+// ith value of 1 means ith latest video is music
+const likedMusicBinaryHist = makeBinaryArr(LIKED_MUSIC_HIST, TOTAL_LIKED);
+
+/**
+ * converts an array to binary arr
+ * @param {number[]} arr values will be indices that hold 1 in output arr
+ * @param {number[]} total size of output arr
+ * @returns binary arr of size total where ith value 1
+ * corresponds to value i in input arr
+ */
+function makeBinaryArr(arr, total) {
+  // first fill with 0s then 1s
+  const binaryArr = [...Array(total)].fill(0);
+  for (const heatIndex of arr) {
+    binaryArr[heatIndex] = 1;
+  }
+  return binaryArr;
+}
+
+/**
+ * turns array of data into heat map
+ * @param {number[]} data array of all 0s/1s heat map data
```

Comment:
This describes the type but not what "data" is. Instead elaborate that data is the list of days in which music was listened to.